tests/test_redis_aggregator.py

```diff
@@ -18,6 +20,11 @@ def on_market_cycle(self, market_info):
         batch_commands = {}
 
         for device_event in market_info["content"]:
+            if "device_info" not in device_event or device_event["device_info"] is None:
+                continue
+            if key_in_dict_and_not_none(device_event, "grid_stats_tree"):
+                json_grid_tree = json.dumps(device_event["grid_stats_tree"], indent=2)
+                print(json_grid_tree)
```

Comment:
I assume that print is needed here in order to display the outcome to Colin, however, could you please replace it with logging.warning or logging.error ? Just in order to have consistency